packages/augur-core/source/contracts/libraries/token/ERC777BaseToken.sol

```diff
@@ -0,0 +1,140 @@
+    function sendNoHooks(address _to, uint256 _amount, bytes32 _data) public returns (bool) {
+        doSend(msg.sender, msg.sender, _to, _amount, _data, "", true, false);
+        return true;
+    }
+
+    function authorizeOperator(address _operator) public returns (bool) {
+        require(_operator != msg.sender, "Cannot authorize yourself as an operator");
+
+        if (isDefaultOperator[_operator]) {
+            revokedDefaultOperator[_operator][msg.sender] = false;
+        } else {
+            authorizedOperators[_operator][msg.sender] = true;
+        }
+        emit AuthorizedOperator(_operator, msg.sender);
+        return true;
+    }
+
+    function revokeOperator(address _operator) public returns (bool) {
+        require(_operator != msg.sender, "Cannot revoke yourself as an operator");
+
+        if (isDefaultOperator[_operator]) {
+            revokedDefaultOperator[_operator][msg.sender] = true;
+        } else {
+            authorizedOperators[_operator][msg.sender] = false;
+        }
+
+        emit RevokedOperator(_operator, msg.sender);
+        return true;
+    }
+
+    function isOperatorFor(address _operator, address _tokenHolder) public view returns (bool) {
+        return (_operator == _tokenHolder // solium-disable-line operator-whitespace
+            || authorizedOperators[_operator][_tokenHolder]
+            || (isDefaultOperator[_operator] && !revokedDefaultOperator[_operator][_tokenHolder]));
+    }
+
+    function operatorSend(address _from, address _to, uint256 _amount, bytes32 _data, bytes32 _operatorData) public returns (bool) {
+        require(isOperatorFor(msg.sender, _from), "Not an operator");
+        doSend(msg.sender, _from, _to, _amount, _data, _operatorData, true, true);
+        return true;
+    }
+
+    function doSend(address _operator, address _from, address _to, uint256 _amount, bytes32 _data, bytes32 _operatorData, bool _preventLocking, bool _callHooks) internal returns (bool) {
+        if (_callHooks) {
+            callSender(_operator, _from, _to, _amount, _data, _operatorData);
+        }
+
+        require(_to != address(0), "Cannot send to 0x0");
+        require(balances[_from] >= _amount, "Not enough funds");
+
+        balances[_from] = balances[_from].sub(_amount);
+        balances[_to] = balances[_to].add(_amount);
+
+        if (_callHooks) {
+            callRecipient(_operator, _from, _to, _amount, _data, _operatorData, _preventLocking);
+        }
+
+        emit Sent(_operator, _from, _to, _amount, _data, _operatorData);
+        return true;
+    }
+
+    function doBurn(address _operator, address _tokenHolder, uint256 _amount, bytes32 _data, bytes32 _operatorData) internal returns (bool) {
+        callSender(_operator, _tokenHolder, 0x0, _amount, _data, _operatorData);
```

Comment:
Will we need a burnNoHooks took too? I don't think we do, because you are never burning another user's tokens in the context of your own transaction, just wanted to bring it up, though

Reply:
Yea, shouldn't be needed. Technically I think even for the sendNoHooks it _could_ just skip the receiver hook for our case but that seems really weird as opposed to just not calling the 820 methods at all

Reply:
I believe that is true; only receive hooks are a vulnerability, but the way it is done looks good, not calling either.